app/scripts/views/settings/clients.js

```diff
@@ -129,48 +128,44 @@ define(function (require, exports, module) {
       });
     },
 
-    _onDisconnectDevice: function (event) {
-      this.logViewEvent('disconnect');
-      var deviceId = $(event.currentTarget).attr('data-id');
-      this._destroyDevice(deviceId);
+    _onDisconnectClient: function (event) {
+      this._attachedClients.removeClient($(event.currentTarget).data('id'), this.user)
```

Comment:
I am wondering if `_attachedClients` needs to be part of this transaction. `attachedClients.removeClient` doesn't actually directly remove the client. The removal occurs when `client.destroy()` is called, which triggers a `destroy` message, which the Collection listens for and removes it.

Instead of going through `this._attachedClient.removeClient`, I think you could do something like:

```js
const client = this._attachedClients.get($(event.currentTarget).data('id')
this.user.destroyAccountClient(this.getCurrentAccount(), client)
  .then(() => ...
```

That should still cause the same sequence of events to happen. One less function needs to be in the Collection. The Collection ends up reacting to the model being destroyed, which causes the ClientsView to update based on the Collection being updated.

I'm not sure. I've probably wasted more time thinking about this than it's worth.